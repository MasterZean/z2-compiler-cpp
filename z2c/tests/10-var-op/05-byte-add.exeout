{
	var a: Byte = Byte{0}
}
{
	var a: Byte = Byte{0}
}
{
	var a: Byte = Byte{0}
}
{
	var a: Short = Short{0}
}
{
	var a: Word = Word{0}
}
{
	var a: Int = Int{0}
}
{
	var a: DWord = DWord{0}
}
{
	var a: Long = Long{0}
}
{
	var a: QWord = QWord{0}
}
{
	var a: Float = Float{0}
}
{
	var a: Double = Double{0}
}
{
	var a: Char = Char{0}
}
{
	var a: PtrSize = PtrSize{0}
}
