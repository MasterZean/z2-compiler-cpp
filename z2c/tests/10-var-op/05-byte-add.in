{ val a = Byte{} + Bool{} }
{ val a = Byte{} + Small{} }
{ val a = Byte{} + Byte{} }
{ val a = Byte{} + Short{} }
{ val a = Byte{} + Word{} }
{ val a = Byte{} + Int{} }
{ val a = Byte{} + DWord{} }
{ val a = Byte{} + Long{} }
{ val a = Byte{} + QWord{} }
{ val a = Byte{} + Float{} }
{ val a = Byte{} + Double{} }
{ val a = Byte{} + Char{} }
{ val a = Byte{} + PtrSize{} }
