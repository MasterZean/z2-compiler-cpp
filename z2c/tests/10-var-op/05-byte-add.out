{
	uint8 a = 0u + false;
}
{
	uint8 a = 0u + 0;
}
{
	uint8 a = 0u + 0u;
}
{
	int16 a = 0u + 0;
}
{
	uint16 a = 0u + 0u;
}
{
	int32 a = 0u + 0;
}
{
	uint32 a = 0u + 0u;
}
{
	int64 a = 0u + 0ll;
}
{
	uint64 a = 0u + 0ull;
}
{
	float a = 0u + 0.0f;
}
{
	double a = 0u + 0.0;
}
{
	uint32 a = 0u + '\0';
}
{
	size_t a = 0u + 0u;
}
